aip/general/0236.md

```diff
@@ -0,0 +1,351 @@
+  `ACTIVE`.
+- If the method is called on an experiment with the rules representing a no-op,
+  then the system **must** preview the deletion of the live policy.
+
+#### stop
+
+```proto
+// Stops previewing a PolicyExperiment. This triggers the system to stop
+// generating logs to evaluate the PolicyExperiment.
+rpc StopPolicyExperiment(StopPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:stop"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "PolicyExperiment"
+    metadata_type: "StopPolicyExperimentMetadata"
+  };
+}
+
+message StopPolicyExperimentRequest {
+  // The name of the PolicyExperiment.
+  string name = 1;
+}
+
+message StopPolicyExperimentMetadata {}
+```
+
+- This custom method is required.
+- This method **must** trigger the system to stop generating logs to preview the
+  experiment.
+- When the method is called, the `state` **must** be set to `INACTIVE` and
+  `stop_time` **must** be set to the current time.
+
+#### commit
+
+The resource **may** expose a new custom method called "commit" to promote an
+experiment. The system copies `policy` from the experiment into the live policy
+and then deletes the experiment.
+
+Declarative clients **may** manually copy fields from an experiment into the
+live policy and then delete the experiment rather than calling "commit" if
+preferrable.
+
+```proto
+// Commits a PolicyExperiment. This copies the PolicyExperiment's policy message
+// to the live policy then deletes the PolicyExperiment.
+rpc CommitPolicyExperiment(CommitPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:commit"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "google.protobuf.Empty"
+    metadata_type: "CommitPolicyExperimentMetadata"
+  };
+}
+
+message CommitPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+  string parent_etag = 3;
+}
+
+message CommitPolicyExperimentMetadata {}
+```
+
+- The method **must** atomically copy `policy` from the experiment into the live
+  policy, and then delete the experiment.
+- If any experiment fails "commit", previewing it **must not** stop, and the
+  live policy **must not** be updated.
+- The method can be called on an experiment in any state.
+- The `etag` **must** match that of the experiment in order for commit to be
+  successful. This is so the user does not commit an unintended version of the
+  experiment.
+  - A `parent_etag` **may** be provided to guarantee that the experiment
+  overwrites a specific version of the live policy.
+- The method is not idempotent and calling it twice on the same experiment 
+  **must** return a 404 NOT_FOUND as the experiment is deleted as part of the
+  first call.
+
+### Live policy API changes
```

Comment:
What doe this header mean? Changes to what? Do you mean to explain how "delete" differs from the standard method, Delete? If so, move above to the "delete" section. A H3 with one H4 isn't very graceful.